### PR TITLE
feat(rvm): implement Azure Policy condition evaluation

### PR DESCRIPTION
**Pull request by anakrish**
Add VM support for Azure Policy's condition operators and allOf/anyOf
short-circuit logic, gated behind cfg(feature = "azure_policy").

Policy conditions (equals, contains, like, match, exists, and their
negations — 21 total) are encoded as a single PolicyCondition
instruction with a PolicyOp sub-opcode rather than bloating the
Instruction enum with 21 variants. The dispatch handles Azure Policy's
quirky comparison semantics: case-insensitive string comparison,
string↔number coercion, null vs undefined distinction, and element-wise
collection membership.

allOf/anyOf blocks use four instructions — LogicalBlockStart,
AllOfNext/AnyOfNext, and LogicalBlockEnd — that wire up a result
register and short-circuit on the first failing (allOf) or passing
(anyOf) child.

Helper functions for case-folded comparison, wildcard/glob matching, and
type coercion live in builtins::azure_policy::helpers.

Two YAML test suites (~2200 lines) exercise the full operator matrix and
the allOf/anyOf control flow.

### CODE REVIEW

File: src/rvm/vm/dispatch.rs [hunk 1]
```diff
@@ -807,6 +807,290 @@ impl RegoVM {
...
+                        }
+                    }
+                    PolicyOp::NotEquals => {
+                        let r = self.get_register(right)?;
+                        if is_undefined(l) {
+                            !matches!(r, Value::Null)
+                        } else {
+                            !case_insensitive_equals(l, r)
+                        }
+                    }
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/rvm/vm/dispatch.rs:929]: This is expected Azure Policy behavior.

File: tests/rvm/vm/suites/azure_policy/policy_allof_anyof.yaml [hunk 2]
```diff
@@ -0,0 +1,678 @@
...
+#   AllOfEnd    — all children passed, set result=true
+#
+# AnyOf: disjunction with lazy short-circuit.
+#   AnyOfStart  — initialize result=false
+#   AnyOfNext   — if child is true, set result=true, jump to AnyOfEnd
+#   AnyOfEnd    — no child matched, result stays false
+#
+# PC semantics: end_pc points to the AllOfEnd/AnyOfEnd instruction.
+# The VM does self.pc = (end_pc as usize).saturating_sub(1) because the
+# main loop does saturating_add(1) after Continue.
```
[the remark this answers is not in the document]
Comment (anakrish) [reply tests/rvm/vm/suites/azure_policy/policy_allof_anyof.yaml]: fixed.

File: tests/rvm/vm/suites/azure_policy/policy_operators.yaml [hunk 3]
```diff
@@ -0,0 +1,1486 @@
...
+    literals: ["hello", []]
+    instructions:
+      - "Load { dest: 1, literal_idx: 0 }"
+      - "Load { dest: 2, literal_idx: 1 }"
+      - "PolicyNotIn { dest: 0, left: 1, right: 2 }"
+      - "Return { value: 0 }"
+    want_result: true
+
+  - note: policy_not_in_not_array
+    description: "right operand is not array → false (not-in a non-array)"
```
[the remark this answers is not in the document]
Comment (anakrish) [reply tests/rvm/vm/suites/azure_policy/policy_operators.yaml]: fixed.

File: tests/rvm/vm/suites/azure_policy/policy_operators.yaml [hunk 4]
```diff
@@ -0,0 +1,1486 @@
...
+
+  - note: policy_match_case_sensitive_fail
+    description: "Match is case-sensitive: 'A' does not match ? when expecting letter check only works for any letter"
+    literals: ["A1", "?#"]
+    instructions:
+      - "Load { dest: 1, literal_idx: 0 }"
+      - "Load { dest: 2, literal_idx: 1 }"
+      - "PolicyMatch { dest: 0, left: 1, right: 2 }"
+      - "Return { value: 0 }"
+    want_result: true
```
[the remark this answers is not in the document]
Comment (anakrish) [reply tests/rvm/vm/suites/azure_policy/policy_operators.yaml]: fixed.

File: src/builtins/azure_policy/helpers.rs [hunk 5]
```diff
@@ -61,6 +221,87 @@ pub fn try_coerce_to_number(s: &str) -> Option<crate::number::Number> {
...
+            if !input_char.is_ascii_digit() {
+                return false;
+            }
+        } else if *pattern_char == '?' {
+            if !input_char.is_ascii_alphabetic() {
+                return false;
+            }
+        } else if input_char != pattern_char {
+            return false;
+        }
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/builtins/azure_policy/helpers.rs]: The `.` wildcard is correct per Azure Policy's `match` operator specification. From the [docs](https://learn.microsoft.com/en-us/azure/governance/policy/concepts/definition-structure-policy-rule#conditions): "provide a hashtag (`#`) to match a digit, question mark (`?`) for a letter, and a dot (`.`) to match any character". No change needed.

File: src/builtins/azure_policy/helpers.rs [hunk 6]
```diff
@@ -61,6 +221,87 @@ pub fn try_coerce_to_number(s: &str) -> Option<crate::number::Number> {
...
+    }
+
+    true
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/builtins/azure_policy/helpers.rs]: fixed.

File: src/rvm/vm/dispatch.rs [hunk 7]
```diff
@@ -807,6 +807,285 @@ impl RegoVM {
...
+            } => {
+                let l = self.get_register(left)?;
+                let result = match op {
+                    PolicyOp::Equals => {
+                        let r = self.get_register(right)?;
+                        if is_undefined(l) {
+                            matches!(r, Value::Null)
+                        } else {
+                            case_insensitive_equals(l, r)
+                        }
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/rvm/vm/dispatch.rs:920]: Intended. Not an issue.

File: src/rvm/vm/dispatch.rs [hunk 8]
```diff
@@ -807,6 +807,285 @@ impl RegoVM {
...
+                    }
+                    PolicyOp::NotEquals => {
+                        let r = self.get_register(right)?;
+                        if is_undefined(l) {
+                            !matches!(r, Value::Null)
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/rvm/vm/dispatch.rs:925]: Same as above. Not an issue.

File: src/rvm/tests/instruction_parser.rs [hunk 9]
```diff
@@ -83,6 +83,40 @@ pub fn parse_instruction(text: &str) -> Result<Instruction> {
...
+            }
+            "PolicyExists" => parse_policy_condition(params_text, PolicyOp::Exists),
+            "ValueConditionGuard" => parse_value_condition_guard(params_text),
+            "PolicyNot" => parse_policy_not(params_text),
+            // AllOf / AnyOf
+            "AllOfStart" => parse_logical_block_start(params_text, LogicalBlockMode::AllOf),
+            "AllOfNext" => parse_allof_next(params_text),
+            "AllOfEnd" => parse_logical_block_end(params_text, LogicalBlockMode::AllOf),
+            "AnyOfStart" => parse_logical_block_start(params_text, LogicalBlockMode::AnyOf),
+            "AnyOfNext" => parse_anyof_next(params_text),
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/rvm/tests/instruction_parser.rs:119]: fixed.

File: src/builtins/azure_policy/helpers.rs [hunk 10]
```diff
@@ -61,6 +216,87 @@ pub fn try_coerce_to_number(s: &str) -> Option<crate::number::Number> {
...
+    let mut pattern_chars = pattern.chars();
+
+    loop {
+        match (input_chars.next(), pattern_chars.next()) {
+            (None, None) => return true,
+            (Some(_), None) | (None, Some(_)) => return false,
+            (Some(input_char), Some(pattern_char)) => {
+                if pattern_char == '.' {
+                    // '.' matches any single character (letter, digit, or special).
+                } else if pattern_char == '#' {
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/builtins/azure_policy/helpers.rs:295]: not an issue. works as according to official docs.

File: src/rvm/program/listing.rs [hunk 11]
```diff
@@ -911,6 +911,16 @@ fn format_instruction_readable(
...
         }
+
+        // Azure Policy & allOf/anyOf instructions — use Display impl
+        instruction @ Instruction::PolicyCondition { .. }
+        | instruction @ Instruction::LogicalBlockStart { .. }
+        | instruction @ Instruction::AllOfNext { .. }
+        | instruction @ Instruction::AnyOfNext { .. }
+        | instruction @ Instruction::LogicalBlockEnd { .. } => {
+            let base = format!("{}{}", indent, instruction);
+            align_comment(&base, "", config.comment_column)
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/rvm/program/listing.rs]: fixed.

File: src/builtins/azure_policy/helpers.rs [hunk 12]
```diff
@@ -61,6 +216,87 @@ pub fn try_coerce_to_number(s: &str) -> Option<crate::number::Number> {
...
+                    if !input_char.is_ascii_digit() {
+                        return false;
+                    }
+                } else if pattern_char == '?' {
+                    if !input_char.is_ascii_alphabetic() {
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/builtins/azure_policy/helpers.rs:300]: `#` is any single digit and ? is any one letter - both in ASCII. Using unicode may not be correct.

File: src/rvm/tests/instruction_parser.rs [hunk 13]
```diff
@@ -93,6 +128,14 @@ pub fn parse_instruction(text: &str) -> Result<Instruction> {
...
             "DestructuringSuccess" => Ok(Instruction::DestructuringSuccess {}),
             "ComprehensionEnd" => Ok(Instruction::ComprehensionEnd {}),
+            "AnyOfEnd" => Ok(
+                parse_logical_block_end(text, LogicalBlockMode::AnyOf).unwrap_or(
+                    Instruction::LogicalBlockEnd {
+                        mode: LogicalBlockMode::AnyOf,
+                        result: 0,
+                    },
+                ),
+            ),
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/rvm/tests/instruction_parser.rs]: fixed.

File: src/builtins/azure_policy/helpers.rs [hunk 14]
```diff
@@ -61,6 +216,87 @@ pub fn try_coerce_to_number(s: &str) -> Option<crate::number::Number> {
...
+    let mut star_index: Option<usize> = None;
+    let mut match_index = 0_usize;
+
+    while input_index < input.len() {
+        let pat_byte = pattern.get(pattern_index).copied();
+        let inp_byte = input.get(input_index).copied();
+
+        if matches!(pat_byte, Some(b'?')) || (pat_byte.is_some() && pat_byte == inp_byte) {
+            input_index = input_index.saturating_add(1);
+            pattern_index = pattern_index.saturating_add(1);
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/builtins/azure_policy/helpers.rs]: already explained.

File: src/builtins/azure_policy/helpers.rs [hunk 15]
```diff
@@ -61,6 +221,87 @@ pub fn try_coerce_to_number(s: &str) -> Option<crate::number::Number> {
...
+        } else if pat_byte == Some(b'*') {
+            star_index = Some(pattern_index);
+            match_index = input_index;
+            pattern_index = pattern_index.saturating_add(1);
+        } else if let Some(star) = star_index {
+            pattern_index = star.saturating_add(1);
+            match_index = match_index.saturating_add(1);
+            input_index = match_index;
+        } else {
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/builtins/azure_policy/helpers.rs:272]: fixed by operating on char instead of u8.

File: src/builtins/azure_policy/helpers.rs [hunk 16]
```diff
@@ -61,6 +216,87 @@ pub fn try_coerce_to_number(s: &str) -> Option<crate::number::Number> {
...
+
+    match_question_hash_pattern(&input, &pattern)
+}
+
+pub fn match_like_pattern_ci(input: &str, pattern: &str) -> bool {
+    wildcard_match(input.as_bytes(), pattern.as_bytes())
+}
+
+fn wildcard_match(input: &[u8], pattern: &[u8]) -> bool {
+    let (mut input_index, mut pattern_index) = (0_usize, 0_usize);
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/builtins/azure_policy/helpers.rs]: fixed by operating on char instead of u8.

File: tests/rvm/vm/suites/azure_policy/policy_operators.yaml [hunk 17]
```diff
@@ -0,0 +1,1486 @@
...
+  - note: policy_in_undefined_left
+    literals: [["Hello"]]
+    instructions:
+      - "Load { dest: 2, literal_idx: 0 }"
+      - "PolicyIn { dest: 0, left: 1, right: 2 }"
+      - "Return { value: 0 }"
+    want_result: false
+
+  - note: policy_in_not_array
+    description: "right operand is not array/set → false"
```
[the remark this answers is not in the document]
Comment (anakrish) [reply tests/rvm/vm/suites/azure_policy/policy_operators.yaml]: already fixed.

File: src/rvm/tests/instruction_parser.rs [hunk 18]
```diff
@@ -93,6 +128,10 @@ pub fn parse_instruction(text: &str) -> Result<Instruction> {
             "RuleReturn" => Ok(Instruction::RuleReturn {}),
             "DestructuringSuccess" => Ok(Instruction::DestructuringSuccess {}),
             "ComprehensionEnd" => Ok(Instruction::ComprehensionEnd {}),
+            "AnyOfEnd" => Ok(Instruction::LogicalBlockEnd {
+                mode: LogicalBlockMode::AnyOf,
+                result: 0,
+            }),
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/rvm/tests/instruction_parser.rs]: fixed.

File: tests/rvm/vm/suites/azure_policy/policy_operators.yaml [hunk 19]
```diff
@@ -0,0 +1,1486 @@
...
+  # Additional corner cases — PolicyNotEquals
+  # =========================================================================
+
+  - note: policy_not_equals_both_undefined
+    description: "both undefined → case_insensitive_equals returns true, then negate → false. But left is undefined → returns true directly."
+    literals: []
+    instructions:
+      - "PolicyNotEquals { dest: 0, left: 1, right: 2 }"
+      - "Return { value: 0 }"
+    want_result: true
```
[the remark this answers is not in the document]
Comment (anakrish) [reply tests/rvm/vm/suites/azure_policy/policy_operators.yaml:846]: fixed.

File: tests/rvm/vm/suites/azure_policy/policy_operators.yaml [hunk 20]
```diff
@@ -0,0 +1,1486 @@
...
+
+  - note: policy_greater_equal_numbers
+    literals: [5, 5]
+    instructions:
+      - "Load { dest: 1, literal_idx: 0 }"
+      - "Load { dest: 2, literal_idx: 1 }"
+      - "PolicyGreater { dest: 0, left: 1, right: 2 }"
+      - "Return { value: 0 }"
+    want_result: false
```
[the remark this answers is not in the document]
Comment (anakrish) [reply tests/rvm/vm/suites/azure_policy/policy_operators.yaml]: fixed.

File: tests/rvm/vm/suites/azure_policy/policy_operators.yaml [hunk 21]
```diff
@@ -0,0 +1,1487 @@
+# Azure Policy condition operator instruction tests
+#
+# Tests for all 20 condition operators + PolicyNot.
+# Each operator implements Azure Policy semantics (case-insensitive,
+# type coercion, undefined-as-missing-field behavior).
+
+cases:
+  # =========================================================================
```
[the remark this answers is not in the document]
Comment (anakrish) [reply tests/rvm/vm/suites/azure_policy/policy_operators.yaml]: fixed.

File: src/rvm/vm/dispatch.rs [hunk 22]
```diff
@@ -807,6 +807,285 @@ impl RegoVM {
...
+                    PolicyOp::ValueConditionGuard => {
+                        // left = value register, right = condition register
+                        if is_undefined(l) {
+                            self.set_register(dest, Value::Bool(false))?;
+                            return Ok(InstructionOutcome::Continue);
+                        } else {
+                            let c = self.get_register(right)?.clone();
+                            self.set_register(dest, c)?;
+                            return Ok(InstructionOutcome::Continue);
+                        }
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/rvm/vm/dispatch.rs:1041]: fixed.

File: src/builtins/azure_policy/helpers.rs [hunk 23]
```diff
@@ -47,6 +47,161 @@ pub fn as_str(value: &Value) -> Option<&str> {
...
+        left
+    };
+    let right = if matches!(right, Value::Null) {
+        &empty
+    } else {
+        right
+    };
+
+    #[allow(clippy::pattern_type_mismatch)]
+    match (left, right) {
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/builtins/azure_policy/helpers.rs]: fixed.

File: src/rvm/vm/dispatch.rs [hunk 24]
```diff
@@ -807,6 +807,285 @@ impl RegoVM {
...
+            // ARM template contains(object, key) checks key membership.
+            Value::Object(ref map) => map.keys().any(|key| case_insensitive_equals(key, r)),
+            // Coerce non-string scalar LHS (e.g., count result)
+            // to a string only when the RHS is already a string.
+            _ => {
+                if matches!(*r, Value::String(_)) {
+                    coerce_to_string(l).is_some_and(|haystack| {
+                        as_string(r)
+                            .is_some_and(|needle| strings::case_fold::contains(&haystack, &needle))
+                    })
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/rvm/vm/dispatch.rs]: fixed.

File: tests/rvm/vm/suites/azure_policy/policy_operators.yaml [hunk 25]
```diff
@@ -0,0 +1,1521 @@
...
+
+  - note: policy_match_exact
+    literals: ["a1b2", "?#?#"]
+    instructions:
+      - "Load { dest: 1, literal_idx: 0 }"
+      - "Load { dest: 2, literal_idx: 1 }"
+      - "PolicyMatch { dest: 0, left: 1, right: 2 }"
+      - "Return { value: 0 }"
+    want_result: true
+
```
[the remark this answers is not in the document]
Comment (anakrish) [reply tests/rvm/vm/suites/azure_policy/policy_operators.yaml:533]: Added tests.

File: tests/rvm/vm/suites/azure_policy/policy_operators.yaml [hunk 26]
```diff
@@ -0,0 +1,1521 @@
...
+      - "Load { dest: 2, literal_idx: 1 }"
+      - "PolicyNotLike { dest: 0, left: 1, right: 2 }"
+      - "Return { value: 0 }"
+    want_result: false
+
+  # =========================================================================
+  # Additional corner cases — PolicyMatch case sensitivity
+  # =========================================================================
+
+  - note: policy_match_case_sensitive_fail
```
[the remark this answers is not in the document]
Comment (anakrish) [reply tests/rvm/vm/suites/azure_policy/policy_operators.yaml]: fixed.

File: src/builtins/azure_policy/helpers.rs [hunk 27]
```diff
@@ -47,6 +47,152 @@ pub fn as_str(value: &Value) -> Option<&str> {
...
+}
+
+pub fn as_boolish(value: &Value) -> Option<bool> {
+    match *value {
+        Value::Bool(b) => Some(b),
+        Value::String(ref s) => match s.to_ascii_lowercase().as_str() {
+            "true" => Some(true),
+            "false" => Some(false),
+            _ => None,
+        },
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/builtins/azure_policy/helpers.rs]: fixed.

File: tests/rvm/vm/suites/azure_policy/policy_operators.yaml [hunk 28]
```diff
@@ -0,0 +1,1521 @@
...
+      - "Load { dest: 2, literal_idx: 1 }"
+      - "PolicyEquals { dest: 0, left: 1, right: 2 }"
+      - "Return { value: 0 }"
+    want_result: true
+
+  - note: policy_equals_undefined_left
+    description: undefined left operand returns false
+    literals: ["hello"]
+    instructions:
+      - "Load { dest: 2, literal_idx: 0 }"
```
[the remark this answers is not in the document]
Comment (anakrish) [reply tests/rvm/vm/suites/azure_policy/policy_operators.yaml:105]: added tests.

File: src/rvm/vm/dispatch.rs [hunk 29]
```diff
@@ -807,6 +807,284 @@ impl RegoVM {
...
+        instruction: Instruction,
+    ) -> Result<InstructionOutcome> {
+        self.execute_virtual_instruction(program, instruction)
+    }
+
+    /// Check whether `l` "contains" `r` using Azure Policy semantics.
+    ///
+    /// Works on strings (case-insensitive substring), arrays/sets (element
+    /// membership), objects (key membership), and coerces non-string scalars
+    /// to strings when the RHS is a string.
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/rvm/vm/dispatch.rs]: fixed docstring

File: src/rvm/vm/dispatch.rs [hunk 30]
```diff
@@ -807,6 +807,284 @@ impl RegoVM {
...
+        }
+    }
+
+    #[cfg(not(feature = "azure_policy"))]
+    fn execute_policy_instruction(
+        &mut self,
+        program: &Program,
+        instruction: Instruction,
+    ) -> Result<InstructionOutcome> {
+        self.execute_virtual_instruction(program, instruction)
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/rvm/vm/dispatch.rs]: fixed.

File: src/rvm/instructions/types.rs [hunk 31]
```diff
@@ -46,6 +46,97 @@ pub enum ComprehensionMode {
...
+            .get(self as usize)
+            .map_or("UNKNOWN_POLICY_OP", |pair| pair.0)
+    }
+
+    /// Compact name for tabular assembly listings.
+    #[allow(clippy::as_conversions)]
+    pub fn compact_name(self) -> &'static str {
+        Self::NAMES
+            .get(self as usize)
+            .map_or("UNKNOWN_POLICY_OP", |pair| pair.1)
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/rvm/instructions/types.rs]: fixed.

File: src/builtins/azure_policy/helpers.rs [hunk 32]
```diff
@@ -61,6 +211,89 @@ pub fn try_coerce_to_number(s: &str) -> Option<crate::number::Number> {
...
+            pattern_index = star.saturating_add(1);
+            match_index = match_index.saturating_add(1);
+            input_index = match_index;
+        } else {
+            return false;
+        }
+    }
+
+    while pattern.get(pattern_index).copied() == Some('*') {
+        pattern_index = pattern_index.saturating_add(1);
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/builtins/azure_policy/helpers.rs]: fixed.

File: src/rvm/instructions/types.rs [hunk 33]
```diff
@@ -46,6 +46,97 @@ pub enum ComprehensionMode {
...
+        ("POLICY_GREATER", "POLICY_GT"),
+        ("POLICY_GREATER_OR_EQUALS", "POLICY_GE"),
+        ("POLICY_LESS", "POLICY_LT"),
+        ("POLICY_LESS_OR_EQUALS", "POLICY_LE"),
+        ("POLICY_IN", "POLICY_IN"),
+        ("POLICY_NOT_IN", "POLICY_NOT_IN"),
+        ("POLICY_CONTAINS", "POLICY_CONTAINS"),
+        ("POLICY_NOT_CONTAINS", "POLICY_NOT_CONTAINS"),
+        ("POLICY_CONTAINS_KEY", "POLICY_CONTAINS_KEY"),
+        ("POLICY_NOT_CONTAINS_KEY", "POLICY_NOT_CK"),
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/rvm/instructions/types.rs]: already fixed.

File: src/rvm/vm/dispatch.rs [hunk 34]
```diff
@@ -807,6 +807,296 @@ impl RegoVM {
...
+                Ok(InstructionOutcome::Continue)
+            }
+            AllOfNext {
+                check,
+                result: _,
+                end_pc,
+            } => {
+                let val = self.get_register(check)?;
+                if !matches!(val, Value::Bool(true)) {
+                    // Child failed — short-circuit.  Result stays false.
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/rvm/vm/dispatch.rs]: fixed.

File: src/builtins/azure_policy/helpers.rs [hunk 35]
```diff
@@ -61,6 +211,107 @@ pub fn try_coerce_to_number(s: &str) -> Option<crate::number::Number> {
...
+        if let (Some((pc, next_pi)), Some((ic, next_ii))) = (pat, inp) {
+            if pc == '?' || pc == ic {
+                pi = next_pi;
+                ii = next_ii;
+                continue;
+            }
+        } else if let Some(('?', next_pi)) = pat {
+            // '?' with no input char remaining is handled by the outer while condition
+            pi = next_pi;
+            continue;
```
[the remark this answers is not in the document]
Comment (anakrish) [reply src/builtins/azure_policy/helpers.rs]: fixed.

**Review by dpokluda — APPROVED** [review]
